From 5d5a131824910eb008599a7c9053fe65b4fc71cf Mon Sep 17 00:00:00 2001
From: Piotr Gorski <lucjan.lucjanov@gmail.com>
Date: Mon, 9 Mar 2020 22:57:33 +0100
Subject: [PATCH 21/33] Revert "port: block, bfq: fix overwrite of bfq_group
 pointer in bfq_find_set_group()"

This reverts commit 6b2e7b3f83566ed05c3dd60b34063421495b6a56.

Signed-off-by: Piotr Gorski <lucjan.lucjanov@gmail.com>
---
 block/bfq-cgroup.c | 9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

diff --git a/block/bfq-cgroup.c b/block/bfq-cgroup.c
index ff6e1e345690..60adc3a3855c 100644
--- a/block/bfq-cgroup.c
+++ b/block/bfq-cgroup.c
@@ -630,15 +630,14 @@ struct bfq_group *bfq_find_set_group(struct bfq_data *bfqd,
 	 */
 	entity = &bfqg->entity;
 	for_each_entity(entity) {
-		struct bfq_group *curr_bfqg = container_of(entity,
-						struct bfq_group, entity);
+		bfqg = container_of(entity, struct bfq_group, entity);
 		BFQ_BUG_ON(!bfqg);
-		if (curr_bfqg != bfqd->root_group) {
-			parent = bfqg_parent(curr_bfqg);
+		if (bfqg != bfqd->root_group) {
+			parent = bfqg_parent(bfqg);
 			if (!parent)
 				parent = bfqd->root_group;
 			BFQ_BUG_ON(!parent);
-			bfq_group_set_parent(curr_bfqg, parent);
+			bfq_group_set_parent(bfqg, parent);
 		}
 	}
 
-- 
2.26.0.51.ga7d14a4428

